cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -70,7 +70,13 @@ public void create(NamespaceMeta namespaceMeta) throws IOException, ExploreExcep
     createLocation(namespaceMeta);
 
     if (cConf.getBoolean(Constants.Explore.EXPLORE_ENABLED)) {
-      exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      try {
+        exploreFacade.createNamespace(namespaceMeta.getNamespaceId().toId());
+      } catch (ExploreException | SQLException e) {
+        // if we failed to create a namespace in explore then delete the earlier created location for the namespace
+        deleteLocation(namespaceMeta.getNamespaceId());
```

Comment:
I see that you have a check in the distributed class as well, just not sure it is clean to have it at two different places